src/ott/solvers/nn/neuraldual.py

```diff
@@ -220,65 +341,75 @@ def train_neuraldual(
 
     return {"train_logs": train_logs, "valid_logs": valid_logs}
 
-  def get_step_fn(self, train: bool, to_optimize: Literal["f", "g"] = "g"):
-    """Create a one-step training and evaluation function."""
+  def get_step_fn(
+      self, train: bool, to_optimize: Literal["f", "g", "parallel"]
+  ):
+    """Create a parallel training and evaluation function."""
 
     def loss_fn(params_f, params_g, f, g, batch):
-      """Loss function for potential f."""
+      """Loss function for both potentials."""
       # get two distributions
-      source, target = batch["source"], batch["target"]
+      X, Y = batch["source"], batch["target"]
```

Comment:
we usually avoid capital letters for variable names.

Reply:
reverted